independent-projects/bootstrap/runner/src/main/java/io/quarkus/bootstrap/runner/PooledBufferAllocator.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkus.bootstrap.runner;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A thread-safe buffer allocator that pools heap buffers.<br>
+ * It's currently implemented using a concurrent linked stack, but the users shouldn't rely on how it's implemented,
+ * as it's subject to change. Conversely, users can rely on the fact that it's thread-safe.
```

Comment:
I believe that within this library you can safely assume there are no "users" - I'd rather elaborate on explaining the implementation and its design choices, to the benefit of ourselves in the future.